bitstream/src/bitread.rs

```diff
@@ -236,13 +236,13 @@ macro_rules! little_endian_reader {
         impl <'a> BitReadEndian for $name<'a> {
             #[inline]
             fn peek_val(&mut self, n:usize) -> u64 {
-                let v = self.cache & ((1u64 << n) - 1);
+                let v = self.cache & ((1u64.checked_shl(n as u32).unwrap_or(0)).overflowing_sub(1).0);
```

Comment:
shouldn't be saturating_sub?

Reply:
saturating_sub won't wrap around when left side is 0 (n = 64)